Build failure with GCC 15 (defaults to -std=gnu23)

Upcoming GCC 15 defaults to `-std=gnu23` and libsndfile fails to build with it as follows:
```
libtool: compile:  x86_64-pc-linux-gnu-gcc -m32 -mfpmath=sse -DHAVE_CONFIG_H -I. -I/var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2 -I./src -I/var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2/src -I/var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2/include -I./include -O2 -march=native -pipe -flto -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wshadow -Wwrite-strings -Wundef -Wuninitialized -Winit-self -Wno-format-truncation -Wvla -Wbad-function-cast -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Waggregate-return -c /var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2/src/ALAC/alac_decoder.c  -fPIC -DPIC -o src/ALAC/.libs/alac_decoder.o
/var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2/src/ALAC/alac_decoder.c:42:9: error: expected identifier before 'false'
   42 | {       false = 0,
      |         ^~~~~
/var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2/src/ALAC/alac_decoder.c:44:3: error: expected ';', identifier or '(' before 'bool'
   44 | } bool ;
      |   ^~~~
/var/tmp/portage/media-libs/libsndfile-1.2.2-r2/work/libsndfile-1.2.2/src/ALAC/alac_decoder.c:44:3: warning: useless type name in empty declaration
make[2]: *** [Makefile:2629: src/ALAC/alac_decoder.lo] Error 1
```

The issue is bool is a keyword in C23 so redefining it breaks. Originally reported downstream in Gentoo at https://bugs.gentoo.org/943864.

--- a/src/ALAC/alac_decoder.c
+++ b/src/ALAC/alac_decoder.c
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <stddef.h>
+#include <stdbool.h>
 #include <string.h>
 
 #include "alac_codec.h"
@@ -39,11 +40,6 @@
 
 #include "ALACBitUtilities.h"
 #include "EndianPortable.h"
-
-typedef enum
-{	false = 0,
-	true = 1
-} bool ;
 
 // constants/data
 const uint32_t kMaxBitDepth = 32 ;			// max allowed bit depth is 32

--- a/src/ALAC/alac_encoder.c
+++ b/src/ALAC/alac_encoder.c
@@ -32,6 +32,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdbool.h>
 #include <string.h>
 
 #include "sfendian.h"
@@ -45,12 +46,6 @@
 #include "ALACBitUtilities.h"
 #include "ALACAudioTypes.h"
 #include "EndianPortable.h"
-
-typedef enum
-{
-	false = 0,
-	true = 1
-} bool ;
 
 static void	GetConfig (ALAC_ENCODER *p, ALACSpecificConfig * config) ;
 

